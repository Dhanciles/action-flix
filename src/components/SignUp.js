import React, { Component } from 'react'

class SignUp extends Component {
  constructor() {
    super()
    this.state = {
      name: '', 
      email: '', 
      password: ''
    }
  }

  render() {
    const {name, email, password} = this.state
    return (
      <form>
        <input name="name" value={name}/>
        <input name="email" value={email}/>
        <input name="password" value={password}/>
        <button>SUBMIT</button>
      </form> 
    )
  }
}