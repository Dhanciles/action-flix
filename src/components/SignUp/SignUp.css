.form-container {
  background-color: #323740;
  height: 100%;
  width: 100%; 
  display: flex; 
  justify-content: center; 
  align-items: center; 
}

form {
  display: flex; 
  flex-direction: column; 
  justify-content: center;  
  align-items: center;
  background-color: white; 
  width: 500px; 
}

form * {
  margin-bottom: 20px; 
  height: 40px; 
  width: 200px;   
}

.inner-container {
  display: flex; 
  justify-content: center; 
  flex-direction: column; 
  background-color: white;
  margin: 0 auto; 
}

.message-container {
  text-align: center; 
}