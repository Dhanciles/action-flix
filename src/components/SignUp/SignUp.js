import React, { Component } from 'react'
import { Redirect } from 'react-router-dom'
import { getUser } from '../../thunks/signIn.js'
import { createUser } from '../../thunks/createUser.js'
import { connect } from 'react-redux'
import './SignUp.css'

export class SignUp extends Component {
  constructor() {
    super()
    this.state = {
      name: '', 
      email: '', 
      password: '',
      showSignIn: true, 
      disableButton: true
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({
      [name]: value
    }, () => this.enableButtons())
}

enableButtons = () => {
  if(this.state.showSignIn && this.state.email !== '' && this.state.password  !== '') {
    this.setState({
      disableButton: !this.state.disableButton
    })
  } else if (!this.state.showSignIn && this.state.email !== ''  && this.state.password !== '' && this.state.name !== '') {
    this.setState({
      disableButton: !this.state.disableButton
    })
  }
}

  handleSubmit = (e) => {
    e.preventDefault()
    const { name, email, password } = this.state
    if (this.state.showSignIn) {
      this.props.checkUserLogin(email, password)      
    } else {
      this.props.createNewUser(name, email, password)
    }

    this.setState({
      name: '',
      email: '',
      password: ''
    })
  }

  handleSignUp = (e) => {
    e.preventDefault()
    this.setState({showSignIn: !this.state.showSignIn})
  }

  render() {
    const {name, email, password, showSignIn} = this.state
    const signUpBtnName = showSignIn ? 'Sign In' : 'Sign Up'
    const newUserBtnName = showSignIn ? 'New user? Sign up!' : 'Log in'
    const popUpMessage = showSignIn ? 'Please Log In' : 'Create Account'
    const errorMessage = showSignIn ? 'email and password do not match' : 'email already exists'

    if (this.props.user) return <Redirect to='/' />
    return (
      <div className="form-container">
        <div className="inner-container">
          <article className="message-container">
            <h1>Welcome To Action Flix!</h1>
            <h3>{popUpMessage}</h3>
          </article>
          <form onSubmit={ this.handleSubmit }>
        {
          !showSignIn ? <input name="name" value={name} placeholder='Enter name ...' onChange={ this.handleChange } /> : ''
        }
          <input name="email" value={email} placeholder='Enter email ...' onChange={ this.handleChange } className='email-input'/>
          <input name="password"  type="password" value={password} placeholder='Enter password ...' onChange={ this.handleChange } />
          <button className='submit-sign-up' disabled={this.state.disableButton} onClick={this.handleSubmit}>{signUpBtnName}</button>        
          <button className='signin-toggle' onClick={this.handleSignUp}>{newUserBtnName}</button>
          { this.props.hasErrored ? <h3>{errorMessage}</h3> : undefined }
          </form> 
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.userId,
  hasErrored: state.hasErrored
})

export const mapDispatchToProps = (dispatch) => ({
  checkUserLogin: (email, password) => dispatch(getUser(email, password)),
  createNewUser: (name, email, password) => dispatch(createUser(name, email, password))
})

export default connect(mapStateToProps, mapDispatchToProps)(SignUp)

