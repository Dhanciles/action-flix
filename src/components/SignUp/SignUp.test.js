import React from 'react'
import { shallow } from 'enzyme'
import SignUp from './SignUp'

describe('SignUp', () => {
  it('should match the snapshot', () => {
    const showSignIn = true
    const name = 'name'
    const email = 'email'
    const password = 'password'

    const wrapper = shallow(
      <form onSubmit={jest.fn()}>
      {
        !showSignIn ? <input name="name" value={name} onChange={jest.fn()} /> : ''
      }
        <input name="email" value={email} onChange={jest.fn()} />
        <input name="password" value={password} onChange={jest.fn()} />
        <button className='submit-sign-up'>SUBMIT</button>
      </form> 
    )

    expect(wrapper).toMatchSnapshot();
  });
})