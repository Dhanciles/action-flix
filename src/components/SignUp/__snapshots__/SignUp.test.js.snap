// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="inner-container"
      >
        <article
          className="message-container"
        >
          <h1>
            Welcome To Action Flix!
          </h1>
          <h3>
            Please Log In
          </h3>
        </article>
        <form
          onSubmit={[Function]}
        >
          
          <input
            className="email-input"
            name="email"
            onChange={[Function]}
            placeholder="Enter email ..."
            value=""
          />
          <input
            name="password"
            onChange={[Function]}
            placeholder="Enter password ..."
            type="password"
            value=""
          />
          <button
            className="submit-sign-up"
            disabled={true}
            onClick={[Function]}
          >
            Sign In
          </button>
          <button
            className="signin-toggle"
            onClick={[Function]}
          >
            New user? Sign up!
          </button>
        </form>
      </div>,
      "className": "form-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            className="message-container"
          >
            <h1>
              Welcome To Action Flix!
            </h1>
            <h3>
              Please Log In
            </h3>
          </article>,
          <form
            onSubmit={[Function]}
          >
            
            <input
              className="email-input"
              name="email"
              onChange={[Function]}
              placeholder="Enter email ..."
              value=""
            />
            <input
              name="password"
              onChange={[Function]}
              placeholder="Enter password ..."
              type="password"
              value=""
            />
            <button
              className="submit-sign-up"
              disabled={true}
              onClick={[Function]}
            >
              Sign In
            </button>
            <button
              className="signin-toggle"
              onClick={[Function]}
            >
              New user? Sign up!
            </button>
          </form>,
        ],
        "className": "inner-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Welcome To Action Flix!
              </h1>,
              <h3>
                Please Log In
              </h3>,
            ],
            "className": "message-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome To Action Flix!",
              },
              "ref": null,
              "rendered": "Welcome To Action Flix!",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Please Log In",
              },
              "ref": null,
              "rendered": "Please Log In",
              "type": "h3",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              <input
                className="email-input"
                name="email"
                onChange={[Function]}
                placeholder="Enter email ..."
                value=""
              />,
              <input
                name="password"
                onChange={[Function]}
                placeholder="Enter password ..."
                type="password"
                value=""
              />,
              <button
                className="submit-sign-up"
                disabled={true}
                onClick={[Function]}
              >
                Sign In
              </button>,
              <button
                className="signin-toggle"
                onClick={[Function]}
              >
                New user? Sign up!
              </button>,
              undefined,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "email-input",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Enter email ...",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "Enter password ...",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
                "className": "submit-sign-up",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New user? Sign up!",
                "className": "signin-toggle",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "New user? Sign up!",
              "type": "button",
            },
            undefined,
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="inner-container"
        >
          <article
            className="message-container"
          >
            <h1>
              Welcome To Action Flix!
            </h1>
            <h3>
              Please Log In
            </h3>
          </article>
          <form
            onSubmit={[Function]}
          >
            
            <input
              className="email-input"
              name="email"
              onChange={[Function]}
              placeholder="Enter email ..."
              value=""
            />
            <input
              name="password"
              onChange={[Function]}
              placeholder="Enter password ..."
              type="password"
              value=""
            />
            <button
              className="submit-sign-up"
              disabled={true}
              onClick={[Function]}
            >
              Sign In
            </button>
            <button
              className="signin-toggle"
              onClick={[Function]}
            >
              New user? Sign up!
            </button>
          </form>
        </div>,
        "className": "form-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <article
              className="message-container"
            >
              <h1>
                Welcome To Action Flix!
              </h1>
              <h3>
                Please Log In
              </h3>
            </article>,
            <form
              onSubmit={[Function]}
            >
              
              <input
                className="email-input"
                name="email"
                onChange={[Function]}
                placeholder="Enter email ..."
                value=""
              />
              <input
                name="password"
                onChange={[Function]}
                placeholder="Enter password ..."
                type="password"
                value=""
              />
              <button
                className="submit-sign-up"
                disabled={true}
                onClick={[Function]}
              >
                Sign In
              </button>
              <button
                className="signin-toggle"
                onClick={[Function]}
              >
                New user? Sign up!
              </button>
            </form>,
          ],
          "className": "inner-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Welcome To Action Flix!
                </h1>,
                <h3>
                  Please Log In
                </h3>,
              ],
              "className": "message-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Welcome To Action Flix!",
                },
                "ref": null,
                "rendered": "Welcome To Action Flix!",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Please Log In",
                },
                "ref": null,
                "rendered": "Please Log In",
                "type": "h3",
              },
            ],
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "",
                <input
                  className="email-input"
                  name="email"
                  onChange={[Function]}
                  placeholder="Enter email ..."
                  value=""
                />,
                <input
                  name="password"
                  onChange={[Function]}
                  placeholder="Enter password ..."
                  type="password"
                  value=""
                />,
                <button
                  className="submit-sign-up"
                  disabled={true}
                  onClick={[Function]}
                >
                  Sign In
                </button>,
                <button
                  className="signin-toggle"
                  onClick={[Function]}
                >
                  New user? Sign up!
                </button>,
                undefined,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "email-input",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "Enter email ...",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "Enter password ...",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign In",
                  "className": "submit-sign-up",
                  "disabled": true,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Sign In",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New user? Sign up!",
                  "className": "signin-toggle",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "New user? Sign up!",
                "type": "button",
              },
              undefined,
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
